woocommerce/wc-admin
# Add PSR-4 naming checks to PHP linting

This PR adds PSR-4 naming checks to the PHP linting, so that we can catch any PSR-4 violations before committing.

It also fixes an issue with `src/Package.php` not being PSR-4 compliant by moving it to `src/Composer/Package.php`.

### Detailed test instructions:

- Check out branch
- Run `npm run lint:php`
- Verify that a PSR-4 error is found in `src/Features/Navigation/Navigation.php`
- Verify that no PSR-4 error is found in `src/Composer/Package.php`
- Add `error_log( __FILE__ );` to the `init()` method in `src/Composer/Package.php` and in your `../woocommerce/packages/woocommerce-admin/src/Package.php`
- With `woocommerce-admin` active, verify that the error log does not contain the path of the packaged woocommerce-admin in core
- Deactivate `woocommerce-admin` plugin
- Verify that the error log does contain the path of the packaged woocommerce-admin in core



### Changelog Note:



Dev: Add PSR-4 naming checks to PHP linting.
